src/gyakuenki_cpp/node/gyakuenki_cpp_node.cpp

```diff
@@ -54,6 +54,36 @@ GyakuenkiCppNode::GyakuenkiCppNode(
         keisan::Matrix<4, 1> Pc;
         ProjectedObject projected_ball =
           this->ipm->map_object(*message, "base_footprint", Pc);
+
+        rclcpp::Time now = this->node->now();
+
+        if (!ball_initialized_) {
+          ball_ekf_.init(Pc[0][0], Pc[1][0]);
+          last_ball_time_ = now;
+          ball_initialized_ = true;
+          return;
+        }
+
+        double dt = (now - last_ball_time_).seconds();
+        last_ball_time_ = now;
+
+        if (dt <= 0.0) {
+          return;
+        }
+
+        ball_ekf_.predict(dt);
+
+        keisan::Matrix<2,1> z;
+        z[0][0] = Pc[0][0];
+        z[1][0] = Pc[1][0];
+
+        ball_ekf_.update(z);
+
+        auto X = ball_ekf_.getstate();
+
+        projected_ball.position.x = X[0][0];
+        projected_ball.position.y = X[1][0];
+
```

Comment:
Move these logic codes to separate cpp files. Cpp files inside node directory should only contain the node configuration itself (publisher-subscriber). Can refer to [akushon](https://github.com/ichiro-its/akushon/blob/master/src/akushon/node/akushon_node.cpp), [ninshiki_cpp](https://github.com/ichiro-its/ninshiki_cpp/blob/master/src/ninshiki_cpp/node/ninshiki_cpp_node.cpp), or other packages as well.